### PR TITLE
feat: Statistic.Countdown pause and start api

### LINKED ISSUE #26182: [Statistic.Countdown] Would be nice to have a pause/start api.
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
In some cases, countdown might need to be paused and continue later on.

### What does the proposed API look like?
const { Countdown  } = Statistic;
const deadline = 1000 * 30; // mili seconds, not depend on Data.now() 
const ref= useRef(null);
const autoStart = false; // boolean

// once mounted, we can call following API to control the countdown
ref.start()
ref.pause()

Countdown: ref={ref} autoStart={autoStart} value={deadline} onFinish={onFinish}

### PR DESCRIPTION
**Pull request by zhangchen915**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #26182 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      Statistic.Countdown pause and start      |
| 🇨🇳 Chinese |      Statistic.Countdown 增加开始和暂停      |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/statistic/index.en-US.md](https://github.com/zhangchen915/ant-design/blob/26182/components/statistic/index.en-US.md)
[View rendered components/statistic/index.zh-CN.md](https://github.com/zhangchen915/ant-design/blob/26182/components/statistic/index.zh-CN.md)

### CODE REVIEW

**Review by xrkffgg — COMMENTED** [review]
feature branch

**Comment by xrkffgg** [comment]
rebase featrue 后
`git push -f`

**Comment by xrkffgg** [comment]
另外建议改用 hooks

**Comment by zhangchen915** [comment]
rebase 到 featrue 发现 CI 有点问题，改用 hooks 是改成函数式组件吗？

**Comment by xrkffgg** [comment]
嗯，Statistic 已经改了，可以把 Countdown 也改一下

**Comment by zombieJ** [comment]
为啥需要 `autoStart`?

**Comment by zombieJ** [comment]
这样设计 `value` 就不受控了

File: components/statistic/Countdown.tsx [hunk 1]
```diff
@@ -9,79 +9,93 @@ interface CountdownProps extends StatisticProps {
...
 
-class Countdown extends React.Component<CountdownProps, {}> {
-  static defaultProps: Partial<CountdownProps> = {
-    format: 'HH:mm:ss',
+const InternalCountdown: React.ForwardRefRenderFunction<unknown, CountdownProps> = (props, ref) => {
+  const { format = '', autoStart } = props;
+  const [countdownId, setCountdownId] = React.useState<number>();
+  const [pauseTime, setPauseTime] = React.useState<number>();
+  const [totalPauseTime, setTotalPauseTime] = React.useState<number>(0);
+  const [, forceUpdate] = React.useState({});
```
Comment (xrkffgg) [inline components/statistic/Countdown.tsx:24]: forceUpdate 可以等一下 https://github.com/ant-design/ant-design/pull/26270 结果

File: components/statistic/index.en-US.md [hunk 2]
```diff
@@ -39,3 +39,11 @@ Display statistic number.
...
 | value | Set target countdown time | number \| moment | - |  |
 | valueStyle | Set value css style | CSSProperties | - |  |
+| autoStart | Auto start countdown | boolean | true |  |
+
+#### Statistic.Countdown Methods
+
+| Name    | Description     | Version |
+| ------- | --------------- | ------- |
+| start() | Start countdown |         |
+| pause() | Pause countdown |         |
```
Comment (zombieJ) [inline components/statistic/index.en-US.md:49]: 不要提供方法出来，会让 `value` 非受控。提供一个 `disabled` 属性，设置后停止计时器，让用户恢复时重新提供一个新的 `value`。

**Review by zombieJ — CHANGES_REQUESTED** [review]

File: components/statistic/index.en-US.md (hunk 2 above)
Comment (zhangchen915) [reply components/statistic/index.en-US.md:49]: 好的

Comment (afc163) [reply components/statistic/index.en-US.md:49]: paused 如何？

**Comment by zombieJ** [comment]
@zhangchen915

File: components/statistic/index.en-US.md (hunk 2 above)
Comment (hengkx) [reply components/statistic/index.en-US.md:49]: paused 感觉 更符合 这个操作一些

**Comment by xrkffgg** [comment]
有冲突，rebase 下最新的

**Comment by afc163** [comment]
Closing for no further progress, feel free to open new pull request: https://github.com/ant-design/ant-design/pull/26229#discussion_r476109433

**Pull request closed without merging**